src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/IDependenciesChanges.cs

```diff
@@ -31,5 +35,25 @@ public void Deconstruct(out string providerType, out string dependencyId)
             providerType = ProviderType;
             dependencyId = DependencyId;
         }
+
+        public bool Equals(RemovedDependencyIdentity other)
+        {
+            return string.Equals(ProviderType, other.ProviderType, StringComparisons.DependencyProviderTypes) && 
+                   string.Equals(DependencyId, other.DependencyId, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public override bool Equals(object obj) => obj is RemovedDependencyIdentity other && Equals(other);
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                return (StringComparer.OrdinalIgnoreCase.GetHashCode(ProviderType) * 397) ^
```

Comment:
Was this hand generated or did the codefix pick this prime?

Reply:
That was the largest prime I could compute on this piece of scrap paper with a hand-drawn sieve of Eratosthenes. <sup>R# generated it</sup>

Would rather use [`System.HashCode`](https://docs.microsoft.com/en-us/dotnet/api/system.hashcode?view=netcore-2.1).

Reply:
Huh, I guess it makes sense.  Prime is a little small for Fowler–Noll–Vo hash function.  For reference, this is what vanilla VS generates:

```C#
public override int GetHashCode()
{
    var hashCode = -427567219;
    hashCode = hashCode * -1521134295 + StringComparer.OrdinalIgnoreCase.GetHashCode(ProviderType);
    hashCode = hashCode * -1521134295 + StringComparer.OrdinalIgnoreCase.GetHashCode(DependencyId);
    return hashCode;
}
```

---
In reply to: [233276869](https://github.com/dotnet/project-system/pull/4271#discussion_r233276869) [](ancestors = 233276869)

Reply:
I trick is make use of ValueTuple; (ProviderType, DependencyId).GetHashCode().

Reply:
VS generates this if tuples are available in the project that you are generating into.